### ISSUE TITLE
SDF/template image support

### ISSUE THREAD

**Issue by mfazekas** [react-native-mapbox-gl/maps#744]
**Describe the bug**
Mapbox seems to be supporting SDF/template images. This allows black and white images/template images colored using icon-color and icon-halo-color

https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/#paint-symbol-icon-color
https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/#paint-symbol-icon-halo-color

https://github.com/mapbox/mapbox-gl-native-ios/commit/88f91a48ae1b237aa285038feecf5060013baad2

https://github.com/mapbox/mapbox-gl-native/pull/12181/files

**Versions (please complete the following information):**
 - Mapbox Version: 8.0.0rc1

**Additional context**
https://stackoverflow.com/questions/33838802/mapbox-gl-change-icon-color

**Comment by bboure** [comment]
@mfazekas So, SDF icons are currently not supported? 
The doc already mentions that and I was trying to find info on how to use them.
(I want to use `iconColor`)

**Comment by andrei-tofan** [comment]
@mfazekas  We've looked a bit into adding the sdf flag to images and found out that will have to extend the api a bit, now images are passed as key value pairs like this:
```
{
 images: {
  "key": require('path to image'), 
}
}
```

and to be able to pass the sdf flag will have to add support for configuration like this as well:

```
{
 images: {
  "key": {
      image: require('path to image'),
      sdf: true
  }, 
}
}
```
What do you guys think?

**Comment by mfazekas** [comment]
@andrei-tofan yep passing objects should work, another alternative would be:
```es6
<Images
  images={key: require('image.png')}
  sdfImages={key:requir('bwImage.png')}
/>
```

**Comment by jaltin** [comment]
Hi @mfazekas,

Thanks for your feedback on the question my colleague @andrei-tofan posted. 

We would really need the SDF to colour icons in our app. If we would make a PR for the change @andrei-tofan suggested, does it stand a chance of being included in the 8.0.0 release (assuming that you approve of the PR of course), or is it too late to make these type of changes to that release?


Many thanks!

**Comment by mfazekas** [comment]
@jaltin I don't think that it's a backward incompatible change. So we should be able to accept it any time.

We'd be happy to merge that PR!

**Comment by simonmitchell** [comment]
@andrei-tofan @jaltin do you have a fork or any example code to show where you've made these changes? :) We also need tintable icons for our app, but need SDF support to do so! Looks like changes may need to be made to the native code, but I can't see anywhere that the code in the originally linked PRs/commits is called!

**Comment by jaltin** [comment]
@simonmitchell, We are working on a PR for this but ran into some unexpected complications in the iOS. 

@andrei-tofan is working on it but we are not sure if/when we will be able to submit a PR for review. He will comment here again once we know more clearly.

**Comment by simonmitchell** [comment]
Awesome thanks both! Look forward to seeing your work :D

**Comment by mfazekas** [comment]
@jaltin can you put your progress somewhere, so others interested can pick it up?!

**Comment by andrei-tofan** [comment]
@mfazekas We've put a bit of work into this, but we decided not to use the sdf feature, nevertheless I will push the code we've done on a branch and share it here.

**Comment by lukemcgregor** [comment]
Ive been playing with SDF images and the mostly seem to work really well. I have gotten it going buy injecting a sprite url into the mapbox style. 

Heres a demo app
https://github.com/lukemcgregor/mapbox-sdf-style

![demo](https://user-images.githubusercontent.com/1284608/84874814-3e6dda80-b0d9-11ea-99d2-7f9ced33f3ab.gif)

**Comment by nthh** [comment]
@mfazekas Is using require() with SDF still unsupported? Is the best alternative to use nativeAssetImages with sdf? 
EDIT: was able to resolve by using 
`{ uri: Image.resolveAssetSource(require('path/to/asset.png')).uri, sdf: true }`
